### PR TITLE
Fixed return type hints detection for abstract functions

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: CodeSniffer/Tokenizers/PHP.php [hunk 1]
```diff
@@ -917,62 +917,76 @@ public function processAdditional(&$tokens, $eolChar)
...
+                    if ($tokens[$x]['code'] === T_COLON) {
+                        for ($y = ($x + 1); $y < $numTokens; $y++) {
+                            if (isset(PHP_CodeSniffer_Tokens::$emptyTokens[$tokens[$y]['code']]) === true) {
+                                continue;
                             }
 
-                            $tokens[$x]['code'] = T_RETURN_TYPE;
-                            $tokens[$x]['type'] = 'T_RETURN_TYPE';
-                        }
+                            if ($tokens[$y]['code'] === T_STRING || $tokens[$y]['code'] === T_ARRAY) {
```
Comment (ondrejmirtes) [inline CodeSniffer/Tokenizers/PHP.php]: `in_array($tokens[$y]['code'], array(T_STRING, T_ARRAY, T_NS_SEPARATOR), true)`

Comment (aik099) [inline CodeSniffer/Tokenizers/PHP.php]: Also test needs to be updated to have namspaced return type.

**Comment by kukulich** [comment]
For details see https://github.com/squizlabs/PHP_CodeSniffer/issues/854

**Comment by gsherwood** [comment]
Thanks a lot for providing this fix and tests.

**Pull request merged**